src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -1509,47 +1509,50 @@ private TypeDeclarationSyntax ParseClassOrStructOrInterfaceDeclaration(SyntaxLis
                 }
 
                 var semicolon = TryEatToken(SyntaxKind.SemicolonToken);
+                var modifiersList = (SyntaxList<SyntaxToken>)modifiers.ToList();
+                var membersList = (SyntaxList<MemberDeclarationSyntax>)members;
+                var constraintsList = (SyntaxList<TypeParameterConstraintClauseSyntax>)constraints;
                 switch (classOrStructOrInterface.Kind)
                 {
                     case SyntaxKind.ClassKeyword:
                         return _syntaxFactory.ClassDeclaration(
                             attributes,
-                            modifiers.ToList(),
+                            modifiersList,
                             classOrStructOrInterface,
                             name,
                             typeParameters,
                             baseList,
-                            constraints,
+                            constraintsList,
                             openBrace,
-                            members,
+                            membersList,
                             closeBrace,
                             semicolon);
 
                     case SyntaxKind.StructKeyword:
                         return _syntaxFactory.StructDeclaration(
                             attributes,
-                            modifiers.ToList(),
+                            modifiersList,
                             classOrStructOrInterface,
                             name,
                             typeParameters,
                             baseList,
-                            constraints,
+                            constraintsList,
                             openBrace,
-                            members,
+                            membersList,
                             closeBrace,
                             semicolon);
 
                     case SyntaxKind.InterfaceKeyword:
                         return _syntaxFactory.InterfaceDeclaration(
                             attributes,
-                            modifiers.ToList(),
+                            modifiersList,
                             classOrStructOrInterface,
                             name,
                             typeParameters,
                             baseList,
-                            constraints,
+                            constraintsList,
                             openBrace,
-                            members,
+                            membersList,
```

Comment:
1. @jaredpar it's a super pity this sort of change needs to be manual.  i suppose it's unsafe for the compielr to figure this out due to potential side-effects?

2. shoudl we get rid of those implicit conversions between the builders and the built items?  so that we have to think about this and can potentially get improvements in other areas of the parser.